pkg/cmd/openshift-tests/collect-disk-certificates/collector.go

```diff
@@ -0,0 +1,208 @@
+package collectdiskcertificates
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphanalysis"
+	"github.com/openshift/library-go/pkg/certs/cert-inspection/certgraphapi"
+	"github.com/openshift/origin/pkg/clioptions/iooptions"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type RunCollectDiskCertificatesFlags struct {
+	ConfigFlags *genericclioptions.ConfigFlags
+	OutputFlags *iooptions.OutputFlags
+
+	ArtifactDir string
+	Prefix      string
+	CollectDirs []string
+
+	genericclioptions.IOStreams
+}
+
+func NewRunCollectDiskCertificatesFlags(ioStreams genericclioptions.IOStreams) *RunCollectDiskCertificatesFlags {
+	return &RunCollectDiskCertificatesFlags{
+		ConfigFlags: genericclioptions.NewConfigFlags(false),
+		OutputFlags: iooptions.NewOutputOptions(),
+		IOStreams:   ioStreams,
+	}
+}
+
+func NewRunCollectDiskCertificatesCommand(ioStreams genericclioptions.IOStreams) *cobra.Command {
+	f := NewRunCollectDiskCertificatesFlags(ioStreams)
+	cmd := &cobra.Command{
+		Use:   "collect-disk-certificates",
+		Short: "Run disk certificate collector",
+		Long: templates.LongDesc(`
+		Run a colelctor which fetches information about certificates stored on disk
+		`),
+
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancelFn := context.WithCancel(context.Background())
+			defer cancelFn()
+			abortCh := make(chan os.Signal, 2)
+			go func() {
+				<-abortCh
+				fmt.Fprintf(f.ErrOut, "Interrupted, terminating\n")
+				cancelFn()
+
+				sig := <-abortCh
+				fmt.Fprintf(f.ErrOut, "Interrupted twice, exiting (%s)\n", sig)
+				switch sig {
+				case syscall.SIGINT:
+					os.Exit(130)
+				default:
+					os.Exit(0)
+				}
+			}()
+			signal.Notify(abortCh, syscall.SIGINT, syscall.SIGTERM)
+
+			if err := f.Validate(); err != nil {
+				return err
+			}
+
+			o, err := f.ToOptions()
+			if err != nil {
+				return err
+			}
+
+			return o.Run(ctx)
+		},
+	}
+
+	f.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (f *RunCollectDiskCertificatesFlags) AddFlags(flags *pflag.FlagSet) {
+	flags.StringArrayVar(&f.CollectDirs, "collect-dir", f.CollectDirs, "directories to collect certs in")
+	flags.StringVar(&f.Prefix, "prefix", f.Prefix, "directories prefix")
```

Comment:
bad arg name and/or description.  Is this a replacement value maybe?  Why are we replacing this actually, why not just leave the full path?

Reply:
Full path starts with `/rootfs` - where we mount host FS. I suppose we can leave it as is

Reply:
Perhaps "root-mount-point" would be better?